### DISCUSSION TITLE
How to install softwares in chrootless fat client

### DISCUSSION THREAD

**Discussion by Abu-Hadi** [ltsp/ltsp#702] [Q&A] [answered]
hi
i am new to LTSP i have setup a small lab of 50 clients with skolelinux edu 11 bullseye installed on server every thing is working fine but the problem is when in install software on ltsp server (chrootless) and after applying ("sudo debian-edu-ltsp-install --diskless_workstation yes") clients not showing newly installed softwares i have "fat clients" and "chrootless setup".
i need help
regard Abu Hadi

**Comment by alkisg** [comment]
I've never even seen a skolelinux setup; maybe its developers can chime-in here.
What is the output of the following commands?

ltsp info
ls -lha /srv/ltsp /srv/ltsp/images

**Reply by Abu-Hadi** [reply]
![IMG20220416122058](https://user-images.githubusercontent.com/103769904/163666108-a3f34356-d486-4223-bc5b-d554d4b82b2c.jpg)

**Reply by Abu-Hadi** [reply]
Hi sir here is my server detail aa per your request

**Comment by Abu-Hadi** [comment]
![IMG_20220416_122326](https://user-images.githubusercontent.com/103769904/163666021-e27d9ad9-5450-46c5-9551-2085aebeabd3.jpg)

**Comment by alkisg** [comment]
And what's the output of the following commands on an LTSP client? You can also use copy/paste instead of screenshots, whatever's easier for you:

```
cat /proc/cmdline
hostname
```

**Reply by Abu-Hadi** [reply]
![IMG_20220416_123447](https://user-images.githubusercontent.com/103769904/163666315-6746e30f-ce1c-4025-8cfd-d072a7a13c8a.jpg)

**Comment by alkisg** [comment]
This is on the server, not on a client... run these commands on an LTSP client.

**Reply by Abu-Hadi** [reply]
Ok

**Reply by Abu-Hadi** [reply]
![IMG20220416124144](https://user-images.githubusercontent.com/103769904/163668206-4df5b7e0-ebf5-4980-b226-7ea903b3ff95.jpg)

**Reply by Abu-Hadi** [reply]
![IMG20220416132415](https://user-images.githubusercontent.com/103769904/163668235-a53bd690-33d3-4c80-81e6-4f5e5f72e77e.jpg)

**Comment by alkisg** [comment, marked as answer]
Which program did you install that has a different version on the server vs the client?

E.g. show us the output of `dpkg -l that-program` from both the server and the client.

**Reply by Abu-Hadi** [reply]
sir in  Debian Edu 11 i never installed any software related to ltsp all the services including ltsp server,apache,squid, mailserver,etc are preinstalled and configured..you just install os and all the services are ready to start working.but when i install extra software related to education like "qb64" and microsoft office with the support of "wine" these application run prety good on server but not showing on ltps clients.one thing i mention that when i login with admin account on ltsp client mechines appllcations shows and working.
In this OS Debians use Gosa web tool to create and manage users, groups and LDAP related works.

![image](https://user-images.githubusercontent.com/103769904/163707467-3e5b5012-4902-415b-aaa5-552b10382e8a.png)

**Comment by alkisg** [comment]
Applications that are installed via WINE are per-user, not per-PC. They are installed in /home/user/.wine, not in /usr/share. They are only available to the user that installed them.
This WINE fact is not related to LTSP at all. Contact the WINE community and ask them:

> How can I install a WINE program on a PC as administrator, and then login as user1 on the same PC and see it?

I.e. you don't even need to involve LTSP servers or clients in your question.
You can see the same problem if you login as user1 on the LTSP server.
You can see the same problem if you install a WINE program as administrator and then login as user1 on a home PC that doesn't have LTSP installed.

**Reply by Abu-Hadi** [reply]
OK thanks sir you update my  knowledge about "wine" but what about "qb64" application its linux base application but i installed this app with (./setup.sh) command this script compile and install qb64 app.

**Reply by Abu-Hadi** [reply]
if this is the case that some applications are user base not machine base then how i install user base softwares on ltsps clients

**Comment by alkisg** [comment]
> If this is the case that some applications are user base not machine base then how i install user base software on LTSP clients

You can ask in your distribution or WINE or software forums, "How can I install a WINE-based or a ./qb64-setup.sh program so that it's available for all users?"
If you make it work for a home PC, it will also work for LTSP; this question and its answer are not specific to LTSP.

**Reply by Abu-Hadi** [reply]
thank you so much alkisg so nice of you my english is not so good but again thanks you clear my confusion. I got your point.
